### PR TITLE
[init] Add transaction fee warnings

### PR DESCRIPTION
**Pull request by maflcko**
Introduce `HIGH_AMOUNT_WARN_MSG` and `LOW_AMOUNT_WARN_MSG`.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
This will normalize the strings to be easier to translate and also change the output text slightly:

(All pictures before/after)

screenshot from 2016-01-18 22-20-45
screenshot from 2016-01-18 22-07-26

---

screenshot from 2016-01-18 22-14-18
screenshot from 2016-01-18 22-14-08

---

screenshot from 2016-01-18 22-07-45
screenshot from 2016-01-18 22-07-59
screenshot from 2016-01-18 22-08-51

File: src/init.cpp [hunk 1]
```diff
@@ -765,9 +765,9 @@ static std::string ResolveErrMsg(const char * const optname, const std::string&
     return strprintf(_("Cannot resolve -%s address: '%s'"), optname, strBind);
 }
 
-static std::string AmountErrMsg(const char * const optname, const std::string& strValue)
+static std::string AmountErrMsg(const std::string& optname, const std::string& strValue)
 {
-    return strprintf(_("Invalid amount for -%s=<amount>: '%s'"), optname, strValue);
+    return strprintf(_("Invalid amount for %s."), strprintf("-%s (%s=%s)", optname, optname, strValue));
```
Comment (luke-jr) [inline src/init.cpp]: Forces English grammar a bit much?

Comment (MarcoFalke) [inline src/init.cpp]: I don't think you can ask someone to "translate" a collection of characters (`-%s=<amount>: '%s'`) into a language.

Also, this seems to work with other grammar just fine:

screenshot from 2016-01-19 08-43-51


Am I missing something?

Comment (luke-jr) [inline src/init.cpp]: German is not the only language. What's wrong with "Invalid amount for -%s (%s=%s)."?

**Comment by laanwj** [comment]
IMO we should settle this code. What do we really want here?

It's been changed way too many times over recent times, I don't think this is warranted by the importance of these errors. At some point it just introduces unnecessary risk of breakage (we don't have good testing for parameter processing), review work, and forces other patches to rebase for no good reason.

**Comment by MarcoFalke** [comment]
@laanwj There is no high/low fee checks for most of the args right now. Currently you can set e.g. mintxfee arbitrarily without a warning.

File: src/init.cpp (hunk 1 above)
Comment (MarcoFalke) [inline src/init.cpp]: There is nothing wrong but I'd rather not expose three %s to translators, considering that the syntax in the braces does not change. Also, this is only a short note to the user that a specific arg can't be parsed, so not worth to have perfect grammar in every possible language, imo.

Though, I am happy to revert, if that's wanted.

**Comment by jonasschnelli** [comment]
Agree with @laanwj. It's already very heavy-to-follow if you want to see the coupling between vars (const, configurable).

**Comment by MarcoFalke** [comment]
Sorry for the odd diff. Now it should be clearer that it is only adding warnings in the appropriate places.

File: src/init.cpp (hunk 1 above)
Comment (MarcoFalke) [inline src/init.cpp]: Reverted

**Comment by MarcoFalke** [comment]
I will open a replacement pull next week.

**Pull request closed without merging**